### PR TITLE
Fix builders and add equality tests

### PR DESCRIPTION
**Pull request by mc-lovin**

### CODE REVIEW

File: src/test/java/com/squareup/kotlinpoet/ParameterSpecTest.kt [hunk 1]
```diff
@@ -47,4 +47,14 @@ class ParameterSpecTest {
         .build()
     assertThat(parameterSpec.toString()).isEqualTo("`with-hyphen`: kotlin.String")
   }
+
+  @Test fun generalBuilderEqualityTest() {
+    val parameterSpec = ParameterSpec.builder("config", String::class)
+        .addAnnotation(ClassName("com.squareup.kotlinpoet", "RunsOnWindows"))
+        .addModifiers(KModifier.FINAL)
+        .defaultValue("{ttl:500}")
```
Comment (JakeWharton) [inline src/test/java/com/squareup/kotlinpoet/ParameterSpecTest.kt]: This is a weird choice... not that it really matters here.

File: src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt [hunk 2]
```diff
@@ -116,4 +116,15 @@ class TypeAliasSpecTest {
     assertThat(a == b).isTrue()
     assertThat(a.hashCode()).isEqualTo(b.hashCode())
   }
+
+  @Test fun generalBuilderEqualityTest() {
+    val typeParam = TypeVariableName("V")
+    val typeAliasSpec = TypeAliasSpec
+        .builder("Bio", Pair::class.parameterizedBy(String:: class, String:: class))
```
Comment (JakeWharton) [inline src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt]: nit: remove spaces after `::` and before `class`

File: src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt [hunk 3]
```diff
@@ -359,4 +359,35 @@ class FunSpecTest {
...
   }
+
+  @Test fun generalBuilderEqualityTest() {
+    val funSpec = FunSpec.Builder("getConfig")
+        .addKdoc("Fix me")
+        .addAnnotation(AnnotationSpec.builder(SuppressWarnings::class)
+            .build())
+        .addModifiers(KModifier.PROTECTED)
+        .addTypeVariable(TypeVariableName("T"))
+        .receiver(String:: class)
```
Comment (JakeWharton) [inline src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: nit: remove space after `::` and before `class`

File: src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt [hunk 4]
```diff
@@ -673,4 +673,19 @@ class FileSpecTest {
         |
         |""".trimMargin())
   }
+
+  @Test fun generalBuilderEqualityTest() {
+    val source = FileSpec.builder("com.squareup.tacos", "Taco")
+        .addAnnotation(JvmMultifileClass::class)
+        .addComment("Generated %L by KotlinPoet. DO NOT EDIT!", "2015-01-13")
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt]: nit: %L feels redundant here

Comment (mc-lovin) [reply src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt]: will remove in a follow up

**Pull request merged**